purchases/src/main/java/com/revenuecat/purchases/interfaces/ReceiveCustomerInfoCallback.kt

```diff
@@ -0,0 +1,38 @@
+//  Purchases
+//
+//  Copyright © 2019 RevenueCat, Inc. All rights reserved.
+//
+package com.revenuecat.purchases.interfaces
+
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.PurchaserInfo
+import com.revenuecat.purchases.PurchasesError
+
+/**
+ * Interface to be implemented when making calls that return a [CustomerInfo]
+ */
+interface ReceiveCustomerInfoCallback {
+    /**
+     * Will be called after the call has completed.
+     * @param customerInfo [CustomerInfo] class sent back when the call has completed
+     */
+    fun onReceived(customerInfo: CustomerInfo)
+
+    /**
+     * Will be called after the call has completed with an error.
+     * @param error A [PurchasesError] containing the reason for the failure of the call
+     */
+    fun onError(error: PurchasesError)
+}
+
+fun ReceivePurchaserInfoListener.toReceiveCustomerInfoCallback(): ReceiveCustomerInfoCallback {
+    return object : ReceiveCustomerInfoCallback {
+        override fun onReceived(customerInfo: CustomerInfo) {
+            this@toReceiveCustomerInfoCallback.onReceived(PurchaserInfo(customerInfo))
```

Comment:
surprised you need the @ here? i find it a little confusing, shouldn't "this" but the ReceivePurchaserInfoListener?

Reply:
If we do `this` here, it will call itself (the `onReceived` in `ReceiveCustomerInfoCallback`), we want to call the `onReceived` of the extension function's receiver. `this@toReceiveCustomerInfoCallback` is the syntax to do that afaik